### ISSUE TITLE
trackRating doesn't get refreshed when rating a song

### ISSUE THREAD

**Issue by fb87** [eonpatapon/gnome-shell-extensions-mediaplayer#303]
I use this for displaying the rating, so either the variable or the indicator text doesn't get refreshed. This doesn't even work when I skip backwards and forwards again. Could this be fixed?

**Comment by JasonLG1979** [comment]
What player? It works fine with every player player I've tested it with that properly handles ratings. It's pretty MPRIS2 spec compliant. It's up to the media player to update the metadata when the ratings change. No signal no update.

Metadata is handled here:
https://github.com/eonpatapon/gnome-shell-extensions-mediaplayer/blob/master/src/player.js#L267-L268

And here:
https://github.com/eonpatapon/gnome-shell-extensions-mediaplayer/blob/master/src/player.js#L400-L422

From there track ratings are handled here:
https://github.com/eonpatapon/gnome-shell-extensions-mediaplayer/blob/834c57a82a3a91c3580f80339ae82c8d38705566/src/ui.js#L216-L224

and here:
https://github.com/eonpatapon/gnome-shell-extensions-mediaplayer/blob/master/src/widget.js#L242-L454

**Comment by fb87** [comment]
The stars in the popover persist after clicking them, but the {trackRating} in the tray text doesn't get updated – I think this is an internal issue…? No new signal should be needed, since the extension causes the change and thus knows this.

**Comment by JasonLG1979** [comment]
Yep there's something going on here. I'm not sure why but here the rating is multiplied by 5?
https://github.com/eonpatapon/gnome-shell-extensions-mediaplayer/blob/ad5f05d0fb6bbe5b9aa86bfdb085d4fe924741fe/src/player.js#L432

**Comment by JasonLG1979** [comment]
I was under the impression that pretty much all players still followed the old MPRIS 1 spec of 0 - 5?

**Comment by JasonLG1979** [comment]
Maybe check to see if the rating is an int or float and act accordingly?

**Comment by eonpatapon** [comment]
I can't remember of any player that did not follow this but lollypop is quite new.

Looking at lollypop code:
```
            self._metadata["xesam:userRating"] =
                Lp().player.current_track.get_popularity() / 5
```

I'm not sure this is the issue here

**Comment by JasonLG1979** [comment]
Yep I don't know what get_popularity() returns. I asked for clarification in the Lollypop bug report.

**Comment by fbruetting** [comment]
Doesn't the MPRIS interface specify the current used version? This would clear out all deviation.

**Comment by JasonLG1979** [comment]
This extension only works with MPRIS2 but that doesn't mean players(or applets for that matter) actually follow the spec
